### PR TITLE
XGBoost linear model support 

### PR DESCRIPTION
**Pull request by izeigerman**
The follow-up PR to https://github.com/BayesWitnesses/m2cgen/pull/133. The `class_idx` argument looks pretty ugly in the `_assemble_single_output`, but so far I couldn't come up with a better name.

### CODE REVIEW

**Comment by izeigerman** [comment]
@StrikerRUS Do you think `split_idx` will be more appropriate?

**Comment by izeigerman** [comment]
Ended up renaming to `split_idx`

**Comment by StrikerRUS** [comment]
Thank you very much for rewriting that PR! Looks very neat at the first glance. I'll take a closer look shortly.

Can you please add API test for XGBoost linear assembler in addition to e2e one?

**Comment by izeigerman** [comment]
Yes, absolutely! Will do so.

File: m2cgen/assemblers/boosting.py [hunk 1]
```diff
@@ -263,11 +314,11 @@ def _count_leaves(tree):
         return num_leaves
 
 
-def _split_trees_by_classes(trees, n_classes):
+def _split_estimators_by_classes(values, n_classes):
```
Comment (StrikerRUS) [inline m2cgen/assemblers/boosting.py]: I'm not sure, is it technically correct to call coefficients "estimators"?..

File: tests/assemblers/test_xgboost.py [hunk 2]
```diff
@@ -10,7 +10,7 @@ def test_binary_classification():
                                       max_depth=1)
     utils.train_model_classification_binary(estimator)
 
-    assembler = assemblers.XGBoostModelAssembler(estimator)
+    assembler = assemblers.XGBoostTreeModelAssembler(estimator)
```
Comment (StrikerRUS) [inline tests/assemblers/test_xgboost.py]: I guess it should be `XGBoostModelAssemblerSelector` (here and all other occurrences of assemblers in this file) to increase the code coverage.

File: m2cgen/assemblers/boosting.py (hunk 1 above)
Comment (izeigerman) [reply m2cgen/assemblers/boosting.py]: Would `estimator_params` make more sense?

File: tests/assemblers/test_xgboost.py (hunk 2 above)
Comment (izeigerman) [reply tests/assemblers/test_xgboost.py]: Makes sense. I can do that 👍

File: README.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ pip install m2cgen
 | **SVM** | <ul><li>LinearSVC</li><li>NuSVC</li><li>SVC</li></ul> | <ul><li>LinearSVR</li><li>NuSVR</li><li>SVR</li></ul> |
 | **Tree** | <ul><li>DecisionTreeClassifier</li><li>ExtraTreeClassifier</li></ul> | <ul><li>DecisionTreeRegressor</li><li>ExtraTreeRegressor</li></ul> |
 | **Random Forest** | <ul><li>ExtraTreesClassifier</li><li>LGBMClassifier(rf booster only)</li><li>RandomForestClassifier</li></ul> | <ul><li>ExtraTreesRegressor</li><li>LGBMRegressor(rf booster only)</li><li>RandomForestRegressor</li></ul> |
-| **Boosting** | <ul><li>LGBMClassifier(gbdt/dart/goss booster only)</li><li>XGBClassifier(gbtree booster only)</li><ul> | <ul><li>LGBMRegressor(gbdt/dart/goss booster only)</li><li>XGBRegressor(gbtree booster only)</li></ul> |
+| **Boosting** | <ul><li>LGBMClassifier(gbdt/dart/goss booster only)</li><li>XGBClassifier(gbtree/gblinear booster only)</li><ul> | <ul><li>LGBMRegressor(gbdt/dart/goss booster only)</li><li>XGBRegressor(gbtree booster only)</li></ul> |
```
[the remark this answers is not in the document]
Comment (izeigerman) [reply README.md]: Oops, missed that one, thanks!

File: m2cgen/assemblers/boosting.py (hunk 1 above)
Comment (StrikerRUS) [reply m2cgen/assemblers/boosting.py]: I'd prefer something even more neutral like "values", but "estimator_params" is rather good too.

**Pull request merged**